spec/controllers/bookings_controller_spec.rb

```diff
@@ -0,0 +1,238 @@
+      should redirect_to bookings_path
+      expect(Booking.exists?(quantity: 1, item: @item,
+                             start_time: DateTime.new(2000, 1, 1, 9), end_time: DateTime.new(2000, 1, 15, 9) ) ).to be true
+    end
+
+    it 'should redirect to bookings_path and create new booking even with existing non-overlapping bookings' do
+      post :create, params: { booking:
+                            { item_id: @item.id,
+                              quantity: 1, 
+                              start_time: DateTime.new(2000, 1, 1, 9),
+                              end_time: DateTime.new(2000, 1, 18, 9) } }
+      should redirect_to bookings_path
+      expect(Booking.exists?(quantity: 1, item: @item,
+                             start_time: DateTime.new(2000, 1, 1, 9), end_time: DateTime.new(2000, 1, 18, 9) ) ).to be true
+    end
+
+    it 'should redirect_to new_booking_path and not create new booking' do
+      post :create, params: { booking:
+                            { item_id: @item.id,
+                              description: 'Failed',
+                              quantity: 2, 
+                              start_time: DateTime.new(2000, 1, 1, 9),
+                              end_time: DateTime.new(2000, 1, 15, 9) } }  
+      assert_template :new
+      expect(Booking.exists?(description: 'Failed')).to be false
+
+      post :create, params: { booking:
+                            { item_id: @item.id,
+                              description: 'Failed',
+                              quantity: 1, 
+                              start_time: DateTime.new(2000, 1, 17, 9),
+                              end_time: DateTime.new(2000, 1, 23, 9) } }  
+      assert_template :new
+      expect(Booking.exists?(description: 'Failed')).to be false
+
+      post :create, params: { booking:
+                            { item_id: @item.id,
+                              description: 'Failed',
+                              quantity: 1, 
+                              start_time: DateTime.new(2000, 2, 18, 9),
+                              end_time: DateTime.new(2000, 2, 23, 9) } }  
+      assert_template :new
+      expect(Booking.exists?(description: 'Failed')).to be false
+    end
+
+    after do
+      Timecop.return
+    end
+  end  
+
+  describe 'GET bookings#new' do
+    before do
+      sign_in create(:user)
+      get :new
+    end
+
+    it { should respond_with :ok }
+  end
+
+  describe 'PATCH bookings#update' do
+    before do
+      sign_in create(:user)
+
+      freeze_time = DateTime.new(2000, 1, 1, 0)
+      Timecop.freeze(freeze_time)
+
+      @booking = create(:booking)
+    end
+
+    it 'should change the description and redirect' do
+      expect do
+        patch :update, params: { id: @booking.id, booking: {description: "I'm super duper Rich"} }
+      end.to change { Booking.find(@booking.id).description }
+        .to("I'm super duper Rich")
+      should redirect_to bookings_path
+    end
+
```

Comment:
Add a test changing to a quantity greater than the item.quantity and make sure it doesn't work